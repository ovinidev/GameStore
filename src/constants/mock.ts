import { Game } from "@interfaces/games";

export const GAME_MOCK: Game = {
	title: "Game Beta",
	description: "A thrilling game",
	genre: "Action",
	id: 1,
	image: "image-url",
	platform: "PC",
	price: 60,
	release_date: "2024-08-01",
};

export const GAMES_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];

export const CART_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];
