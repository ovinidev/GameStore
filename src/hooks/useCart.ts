import { useAddCart, useDeleteCart } from "@mutations/cart";
import { useGetCart } from "@queries/cart";

export const useCart = () => {
	const { data } = useGetCart();
	const { mutateAsync: addToCart } = useAddCart();
	const { mutateAsync: deleteFromCart } = useDeleteCart();

	return {
		addToCart,
		deleteFromCart,
		data,
	};
};
