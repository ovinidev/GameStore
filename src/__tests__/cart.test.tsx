import { CART_MOCK } from "@constants/mock";
import { useCart } from "@hooks/useCart";
import { render } from "@testing-library/react-native";
import Cart from "../app/cart";

jest.mock("expo-router", () => ({
	Stack: {
		Screen: "Screen",
		Navigator: "Navigator",
	},
	useRouter: jest.fn().mockReturnValue({
		push: jest.fn(),
		setParams: jest.fn(),
	}),
}));

jest.mock("@hooks/useCart", () => ({
	useCart: jest.fn().mockReturnValue({
		data: jest.fn(),
	}),
}));

describe("Cart page", () => {
	it("should be able to match the snapshot when cart is not empty", async () => {
		(useCart as jest.Mock).mockReturnValue(CART_MOCK);

		const { toJSON } = render(<Cart />);

		expect(toJSON()).toMatchSnapshot();
	});

	it("should be able to match the snapshot when cart is empty", () => {
		(useCart as jest.Mock).mockReturnValue({
			data: {
				cart: [],
				totalPrice: 0,
			},
		});

		const { toJSON } = render(<Cart />);

		expect(toJSON()).toMatchSnapshot();
	});
});
