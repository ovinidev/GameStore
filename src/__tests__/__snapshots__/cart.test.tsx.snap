// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page should be able to match the snapshot when cart is empty 1`] = `
<View
  className="flex-1 bg-slate-950 p-6"
>
  <Screen
    options={
      {
        "headerRight": [Function],
        "headerTitle": [Function],
      }
    }
  />
  <Text
    className="self-start font-normal text-slate-100 text-xl"
  >
    O carrinho de compras est√° vazio
  </Text>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    className="mt-4 flex flex-row items-center justify-between bg-primary p-2"
  >
    <Text
      className="font-semibold text-slate-100 text-xl"
    >
      Total: R$ 
      0
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-center font-bold text-md text-slate-50"
      >
        Finalizar compra
      </Text>
    </View>
  </View>
</View>
`;

exports[`Cart page should be able to match the snapshot when cart is not empty 1`] = `
<View
  className="flex-1 bg-slate-950 p-6"
>
  <Screen
    options={
      {
        "headerRight": [Function],
        "headerTitle": [Function],
      }
    }
  />
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    className="mt-4 flex flex-row items-center justify-between bg-primary p-2"
  >
    <Text
      className="font-semibold text-slate-100 text-xl"
    >
      Total: R$ 
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-center font-bold text-md text-slate-50"
      >
        Finalizar compra
      </Text>
    </View>
  </View>
</View>
`;
